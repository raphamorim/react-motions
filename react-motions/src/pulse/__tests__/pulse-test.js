import React from 'react';
import { mount } from 'enzyme';
import Pulse from '../pulse';

describe('Pulse', () => {
  it('should render', () => {
    const wrapper = shallow(<Pulse>Pulse</Pulse>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should receive duration prop', () => {
    const wrapper = mount(<Pulse duration={3}>Pulse</Pulse>);

    expect(wrapper.props().duration).toEqual(3);
  });
  it('should receive fillMode prop', () => {
    const wrapper = mount(<Pulse fillMode="both">Pulse</Pulse>);
    expect(wrapper.props().fillMode).toEqual("both");
  });
  it('should render correctly when infinite', () => {
    const wrapper = mount(<Pulse infinite>Pulse</Pulse>);

    expect(wrapper.props().infinite).toBe(true);
  });
});
