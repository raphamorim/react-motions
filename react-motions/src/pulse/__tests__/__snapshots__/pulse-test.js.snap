// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pulse should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pulse
    duration={1}
    fillMode="both"
    infinite={false}
  >
    Pulse
  </Pulse>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Pulse",
      "style": Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "pulse",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "pulse",
      },
    },
    "ref": null,
    "rendered": "Pulse",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Pulse",
        "style": Object {
          "WebkitAnimationDuration": "1s",
          "WebkitAnimationFillMode": "both",
          "WebkitAnimationName": "pulse",
          "animationDuration": "1s",
          "animationFillMode": "both",
          "animationName": "pulse",
        },
      },
      "ref": null,
      "rendered": "Pulse",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
